dhl_sdk/_input_processing.py

```diff
@@ -0,0 +1,718 @@
+                    formatted_inputs[variable.id]["timestamps"] = self.timestamps[
+                        : len(value)
+                    ]
+                    formatted_inputs[variable.id]["steps"] = self.steps[: len(value)]
+                    break
+
+        for variable in input_variables:
+            if variable.id in formatted_inputs:
+                instances[0].append(formatted_inputs[variable.id])
+            else:
+                instances[0].append(None)
+
+        json_data = {"instances": instances, "config": {"startingIndex": 0}}
+
+        return [json_data]
+
+
+def _validate_upstream_timestamps(
+    timestamps: list[Union[int, float]], timestamps_unit: str
+) -> list[Union[int, float]]:
+    """Validate the timestamps for upstream prediction.
+
+    This function performs a series of validations on the provided timestamps to ensure they meet
+    the requirements for upstream prediction tasks.
+
+    Parameters
+    ----------
+    timestamps : list[int]
+        List of timestamps
+
+    """
+    # Validate types
+    if not isinstance(timestamps, list):
+        raise InvalidTimestampsException("Timestamps must be a list of numbers")
+
+    # Validate if length of timestamps
+    if len(timestamps) <= 1:
+        raise InvalidTimestampsException(
+            "Timestamps must be a list of at least 2 values"
+        )
+
+    # Validate if timestamps are valid numeric values
+    if not all(
+        isinstance(value, (int, float)) and math.isfinite(value) for value in timestamps
+    ):
+        raise InvalidTimestampsException(
+            "All values of timestamps must be valid numeric values"
+        )
+
+    # Validate if timestamps are in ascending order and unique
+    if any(a >= b for a, b in zip(timestamps, timestamps[1:])):
+        raise InvalidTimestampsException("Timestamps must be in ascending order")
+
+    # Validate if timestamps are positive (Since they are ordered, just check the first one)
+    if timestamps[0] < 0:
+        raise InvalidTimestampsException("Timestamps must be positive")
+
+    # Convert timestamps to seconds
+    if timestamps_unit.lower() in ("s", "sec", "secs", "seconds"):
+        return timestamps
+    if timestamps_unit.lower() in ("m", "min", "mins", "minutes"):
+        factor = 60
+        return [timestamp * factor for timestamp in timestamps]
+    if timestamps_unit.lower() in ("h", "hour", "hours"):
+        factor = 60 * 60
+        return [timestamp * factor for timestamp in timestamps]
+    if timestamps_unit.lower() in ("d", "day", "days"):
+        factor = 60 * 60 * 24
+        return [timestamp * factor for timestamp in timestamps]
+
+    raise InvalidTimestampsException(
+        f"Invalid timestamps unit '{timestamps_unit}' found."
+    )
+
+
+def _validate_historical_steps(
+    steps: list[Optional[int]], timestamps: list[Union[int, float]]
+):
+    """Validate the steps for historical prediction.
```

Comment:
typehint `-> None` is missing from this function

additionally, here you dont mention the raised exceptions, but in other private methods you do mention the raised exceptions in the docstring, i'd say we include raised exceptions everywhere?